### ISSUE TITLE
Mapping validation within templates

### ISSUE THREAD

**Issue by markwalkom** [elastic/elasticsearch#19532]
I know this is PEBKAC (ie my fault), but we should validate this stuff.

Put a template with mappings, note the (mis)use of `"type" : "flot"`;
```
PUT _template/test
{
  "template": "test",
  "order": 0,
  "mappings": {
    "test": {
      "properties": {
        "floatfield": {
          "type" : "flot"
        }
      }
    }
  }
}
```
Response;
```
{
  "acknowledged": true
}
```

Put a doc;
```
PUT test/test/1
{
  "floatfield": 0.1
}
```

Response;
```
{
   "error": {
      "root_cause": [
         {
            "type": "mapper_parsing_exception",
            "reason": "No handler for type [flot] declared on field [floatfield]"
         }
      ],
      "type": "mapper_parsing_exception",
      "reason": "Failed to parse mapping [test]: No handler for type [flot] declared on field [floatfield]",
      "caused_by": {
         "type": "mapper_parsing_exception",
         "reason": "No handler for type [flot] declared on field [floatfield]"
      }
   },
   "status": 400
}
```

If you do the same thing with a direct mapping call then we catch it straight up.

**Comment by brwe** [comment]
I think this is fixed with https://github.com/elastic/elasticsearch/pull/8802 

In 5.0-alpha4 you will get:


```
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "No handler for type [flot] declared on field [floatfield]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping [test]: No handler for type [flot] declared on field [floatfield]",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "No handler for type [flot] declared on field [floatfield]"
    }
  },
  "status": 400
}
```

**Comment by clintongormley** [comment]
Closed by #8802
